### PR TITLE
Multi-Arch Images, use Docker Build Kit

### PR DESCRIPTION
**Pull request by remuslazar**
## Goal/Incentive

Have Multi-Arch Docker Images so we can use them on arm based architecture, especially on M1 Macs.

## Links

https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/

## Status

IT WORKS!

### CODE REVIEW

**Comment by baschny** [comment]
why do we need this?

**Comment by remuslazar** [comment]
> why do we need this?

My goal is to create multi arch images, so we can use them both in amd64 but also for M1. I thought that just building them on the M1 and push them will be fine, ending with having both archs available on docker. But unfortunately this is not the case; doing so the existing image will get overwritten.

Here is a good tutorial for how to build multi arch with buildx:

https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/

And I thought, I will just try this out.

But this didn't went well, see the error message from the description.

So I will leave this as a draft for now, having M1 images is not really required right now, it was more of less an experiment of mine, I was curios how it will perform on the M1. But because of our old ElasticSearch version requirement this [will not work out anyway](https://cron.slack.com/archives/D027Z2AHM/p1617473388000800).

**Comment by remuslazar** [comment]
@baschny could u please review that? Now everything works; there are two distinct workflows, one for build and test, as before, the other for build and push, using buildx and QEMU64 für an arm64 build. This is insanely slow, see https://github.com/cron-eu/docker-neos/actions/runs/1440284376 but it works!

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -24,9 +27,10 @@ jobs:
         uses: actions/cache@v2
         with:
           path: /tmp/.buildx-cache
-          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          key: ${{ runner.os }}-test-buildx-${{ matrix.php-version }}-${{ matrix.target }}-${{ github.sha }}
```
Comment (baschny) [inline .github/workflows/build-and-test.yml:30]: Can't we drop the `github.sha` in the cache key to be able to reuse the cache even across multiple runs even on different commits? For example in a pull request, every new commit would get a new cache entry, while certain docker layers doesn't need to be rebuild every time - speeding up the process. Docker will not reuse a layer anyways if the content is changed.

**Review by baschny — APPROVED** [review]

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (remuslazar) [reply .github/workflows/build-and-test.yml:30]: this is already the case. You have to use distinct cache keys but in restore-keys we will basically not constrain for the hash. But doing so, having multiple invocations in parallel will not break the cache then.

Comment (remuslazar) [reply .github/workflows/build-and-test.yml:30]: see also https://github.community/t/understanding-cache-action-keys/17503

**Pull request merged**